Explorer/Assets/Scripts/ECS/SceneLifeCycle/SceneDefinition/Components/SceneDefinitionComponent.cs

```diff
@@ -9,61 +8,130 @@
-        public readonly bool IsSDK7;
-        public readonly ParcelMathHelper.SceneGeometry SceneGeometry;
-        public int InternalJobIndex;
+        public int InternalJobIndex { get; set; }
 
-        public SceneDefinitionComponent(SceneEntityDefinition definition, IpfsPath ipfsPath)
+        public SceneDefinitionComponent(
+            SceneEntityDefinition definition,
+            IReadOnlyList<Vector2Int> parcels,
+            IReadOnlyList<ParcelMathHelper.ParcelCorners> parcelsCorners,
+            ParcelMathHelper.SceneGeometry sceneGeometry,
+            IpfsPath ipfsPath, bool isEmpty, bool isSDK7, bool isPortableExperience)
         {
             Definition = definition;
-            ParcelsCorners = new List<ParcelMathHelper.ParcelCorners>(definition.metadata.scene.DecodedParcels.Select(ParcelMathHelper.CalculateCorners));
+            Parcels = parcels;
+            ParcelsCorners = parcelsCorners;
             IpfsPath = ipfsPath;
-            Parcels = definition.metadata.scene.DecodedParcels;
-            IsEmpty = false;
-            IsSDK7 = definition.metadata?.runtimeVersion == "7";
-            SceneGeometry = ParcelMathHelper.CreateSceneGeometry(ParcelsCorners, Definition.metadata.scene.DecodedBase);
+            IsEmpty = isEmpty;
+            IsSDK7 = isSDK7;
+            SceneGeometry = sceneGeometry;
             InternalJobIndex = -1;
+            IsPortableExperience = isPortableExperience;
+        }
+    }
+
+    public static class SceneDefinitionComponentFactory
+    {
+        private static ParcelMathHelper.SceneGeometry cachedPortableExperiencesSceneGeometry;
+        private static List<ParcelMathHelper.ParcelCorners> cachedParcelsCorners;
+
+        public static SceneDefinitionComponent CreateFromDefinition(SceneEntityDefinition definition, IpfsPath ipfsPath)
+        {
+            if (definition.metadata.isPortableExperience) { return CreatePortableExperienceSceneDefinitionComponent(definition, ipfsPath); }
+
+            return CreateSceneDefinitionComponent(definition, definition.metadata.scene.DecodedParcels, ipfsPath, isEmpty: false, isSDK7: definition.metadata.runtimeVersion == "7", isPortableExperience: false);
+        }
+
+        public static SceneDefinitionComponent CreatePortableExperienceSceneDefinitionComponent(SceneEntityDefinition definition, IpfsPath ipfsPath)
+        {
+            if (cachedParcelsCorners == null)
+            {
+                var portableParcels = new List<Vector2Int>();
+
+                for (int i = -150; i < 150; i++)
+                {
+                    for (int j = -150; j < 150; j++) { portableParcels.Add(new Vector2Int(i, j)); }
+                }
+
+                cachedParcelsCorners = new List<ParcelMathHelper.ParcelCorners>(portableParcels.Select(ParcelMathHelper.CalculateCorners));
+                cachedPortableExperiencesSceneGeometry = ParcelMathHelper.CreateSceneGeometry(cachedParcelsCorners, definition.metadata.scene.DecodedBase);
```

Comment:
You don't need to run `CreateSceneGeometry` for -150, 150 - you know the result beforehand and can hardcode it

Reply:
Yess, you are totally right, I was leaving it in case we decided to change the -150, 150 value, but I guess for now we can hardcode it as well and just re-calculate it if needed in the future...